File: src/main/java/frc/robot/Extreme3DProJoystick.java
package frc.robot;

/**
 * Bindable Joystic Class for the Extreme 3D Pro Joystick that comes in the Kit of Parts.
 */
public class Extreme3DProJoystick extends BindableJoystick {

	// Buttons
	public static final int TRIGGER = 1;
	public static final int THUMB_BUTTON = 2;

	public static final int TOP_LEFT_TOP_BUTTON = 5;
	public static final int TOP_RIGHT_TOP_BUTTON = 6;
	public static final int BOTTOM_LEFT_TOP_BUTTON = 3;
	public static final int BOTTOM_RIGHT_TOP_BUTTON = 4;

	public static final int TOP_LEFT_BASE_BUTTON = 7;
	public static final int TOP_RIGHT_BASE_BUTTON = 8;
	public static final int MIDDLE_LEFT_BASE_BUTTON = 9;
	public static final int MIDDLE_RIGHT_BASE_BUTTON = 10;
	public static final int BOTTOM_LEFT_BASE_BUTTON = 11;
	public static final int BOTTOM_RIGHT_BASE_BUTTON = 12;

	public Extreme3DProJoystick(int port) {
		super(port);
	}
}

File: src/main/java/frc/robot/operatorinterface/Extreme3DProJoystick.kt
package frc.robot.operatorinterface

/**
 * Bindable Joystic Class for the Extreme 3D Pro Joystick that comes in the Kit of Parts.
 *
 * @author Brennon Brimhall <brennonbrimhall@provotypes.org
 */
class Extreme3DProJoystick(port: Int) : BindableJoystick(port) {

    /**
     * Buttons for this controller.
     */
    companion object {
        val TRIGGER = 1
        val THUMB_BUTTON = 2

        val TOP_LEFT_TOP_BUTTON = 5
        val TOP_RIGHT_TOP_BUTTON = 6
        val BOTTOM_LEFT_TOP_BUTTON = 3
        val BOTTOM_RIGHT_TOP_BUTTON = 4

        val TOP_LEFT_BASE_BUTTON = 7
        val TOP_RIGHT_BASE_BUTTON = 8
        val MIDDLE_LEFT_BASE_BUTTON = 9
        val MIDDLE_RIGHT_BASE_BUTTON = 10
        val BOTTOM_LEFT_BASE_BUTTON = 11
        val BOTTOM_RIGHT_BASE_BUTTON = 12
    }
}
